### PR TITLE
added Dockerfile and docker-compose

### PR DESCRIPTION
**Pull request by JamesWP**
To build and run the container:
```bash
$ docker-compose up
```
If you have changed source since last build then ...
```bash
$ docker-compose up --build
```
... will rebuild things if required. In actual fact the second is preferable and quick enough if the docker cache is populated.

### CODE REVIEW

**Comment by JamesWP** [comment]
addresses #14

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,22 +11,27 @@ include_directories(".")
 include_directories("exchange")
 include_directories("fxbattle")
 
-# sources
```
Comment (nadvamir) [inline CMakeLists.txt:14]: y no glob?

Comment (JamesWP) [reply CMakeLists.txt:14]: Glob is generally agreed as bad practice in Clark files.

And also... it didn’t work in the docket container for some reason

Comment (JamesWP) [reply CMakeLists.txt:14]: Cmake*

**Review by nadvamir — COMMENTED** [review]
Update README.md with docker-compose as a dependency?

File: CMakeLists.txt (hunk 1 above)
Comment (nadvamir) [reply CMakeLists.txt:14]: Bad practice -- probably, but listing all files can be annoying too :) Will have to live with it I guess

**Comment by nadvamir** [comment]
And also put the running commands in readme to?

**Review by nadvamir — APPROVED** [review]
Thank you!

**Pull request merged**